user/tests/wiring/no_fixture_long_running/cellular.cpp

```diff
@@ -114,7 +114,59 @@ test(CELLULAR_02_device_will_connect_to_the_cloud_when_all_udp_sockets_consumed)
     // Then the device overcomes this socket obstacle and connects to the Cloud
     assertEqual(Particle.connected(), true);
 }
-
 #endif // !HAL_USE_SOCKET_HAL_POSIX
 
+test(CELLULAR_03_device_will_reconnect_to_the_cloud_when_unexpected_modem_brown_out_occurs_within_2min) {
+#if PLATFORM_ID != PLATFORM_ELECTRON
+    skip();
+    return;
+#endif
+    // Serial.println("the device will reconnect to the cloud when unexpected modem brown out occurs within 2min");
+    // Given the device is currently connected to the Cloud
+    connect_to_cloud(6*60*1000);
+    // When the device modem suffers a simulated brown out (modem reset and powered back on with no initialization)
+    digitalWrite(RESET_UC, 0);
+    delay(10000);
+    digitalWrite(RESET_UC, 1);
+    do {
+        // SARA-U2/LISA-U2 50..80us
+        digitalWrite(PWR_UC, 0);
+        delay(50);
+        digitalWrite(PWR_UC, 1);
+        delay(10);
+
+        // SARA-G35 >5ms, LISA-C2 > 150ms, LEON-G2 >5ms, SARA-R4 >= 150ms
+        digitalWrite(PWR_UC, 0);
+        delay(150);
+        digitalWrite(PWR_UC, 1);
+        delay(100);
+    } while (RESP_OK != Cellular.command(1000, "AT\r\n"));
+    // Then the device overcomes this brown out obstacle and reconnects to the Cloud within 2 minutes
+    system_tick_t start = millis();
+    waitFor(Particle.disconnected, 60*1000);
+    assertEqual(Particle.disconnected, true);
+    waitFor(Particle.connected, 2*60*1000 - (millis() - start));
+    assertEqual(Particle.connected(), true);
+}
+
+test(CELLULAR_04_device_will_reconnect_to_the_cloud_when_unexpected_modem_reset_occurs_within_2min) {
+#if PLATFORM_ID != PLATFORM_ELECTRON
+    skip();
+    return;
+#endif
+    // Serial.println("the device will reconnect to the cloud when unexpected modem reset occurs within 2min");
+    // Given the device is currently connected to the Cloud
+    connect_to_cloud(6*60*1000);
+    // When the device modem suffers a simulated brown out (modem reset and powered back on with no initialization)
+    digitalWrite(RESET_UC, 0);
+    delay(10000);
+    digitalWrite(RESET_UC, 1);
```

Comment:
For some reason a comment got lost, so I'll repeat just in case.

Making sure that we are testing for the right thing: SARA R4 will not boot back up if it's reset through `RESET` pin. SARA U2 will boot up automatically.

Reply:
If you look a few lines below this it powers on the modem again.  Perhaps not required for U2, but just in case powering on won't hurt.

Reply:
That's a different test though. There is a power-on sequence in `CELLULAR_03_device_will_reconnect_to_the_cloud_when_unexpected_modem_brown_out_occurs_within_2min`.

Reply:
Gotta fix that comment about brown out on this one.  I'd like to test the case where we power off unexpectedly, and reset was the way to get there easily on R410.  Perhaps I can change this to just power off the modem via the PWR pin.